lib/application.js

```diff
@@ -296,7 +296,9 @@ app.enableAuth = function() {
     var req = ctx.req;
     var Model = method.ctor;
     var modelInstance = ctx.instance;
-    var modelId = modelInstance && modelInstance.id || req.param('id');
+    var modelId = (modelInstance && modelInstance.id) ||
+      (req.params && req.params.id) ||
+      (req.body && req.body.id);
```

Comment:
I put it at the bottom because `req.param()` does indeed check the body first, but the id should not be in the request body. Should both `req.params.id` and `req.body.id` I believe the body should be ignored.

Why not remove `req.body.id`?